PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/SingleOperationInLoopCheck.java

```diff
@@ -0,0 +1,288 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiBlockStatement;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiCodeBlock;
+import com.intellij.psi.PsiDeclarationStatement;
+import com.intellij.psi.PsiDoWhileStatement;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.PsiExpression;
+import com.intellij.psi.PsiExpressionStatement;
+import com.intellij.psi.PsiForStatement;
+import com.intellij.psi.PsiForeachStatement;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiStatement;
+import com.intellij.psi.PsiVariable;
+import com.intellij.psi.PsiWhileStatement;
+import com.intellij.psi.util.PsiTreeUtil;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * Inspection to check if there is a Text Analytics client operation inside a loop.
+ * If a Text Analytics client operation is found inside a loop,
+ * and the API has a batch alternative, a problem will be registered.
+ *
+ * THis is an example of a situation where the inspection should register a problem:
+ *
+ * // Loop through the list of texts and detect the language for each text
+*         1. for (String text : texts) {
+*             DetectedLanguage detectedLanguage = textAnalyticsClient.detectLanguage(text);
+*             System.out.println("Text: " + text + " | Detected Language: " + detectedLanguage.getName() + " | Confidence Score: " + detectedLanguage.getConfidenceScore());
+*         }
+*
+*         // Loop through the list of texts and detect the language for each text
+*        2.  DetectedLanguage detectedLanguage = null;
+*         for (String text : texts) {
+*             textAnalyticsClient.detectLanguage(text);
+ */
+public class SingleOperationInLoopCheck extends LocalInspectionTool {
```

Comment:
i thought to keep it as is because most main classes in the other plugins in this repo are public class as well -- perhaps there's some reasoning behind that